src/Development/IDE/Core/Shake.hs

```diff
@@ -815,6 +815,6 @@ updatePositionMapping IdeState{shakeExtras = ShakeExtras{positionMapping}} Versi
         let uri = toNormalizedUri _uri
         let mappingForUri = Map.findWithDefault Map.empty uri allMappings
         let updatedMapping =
-                Map.insert _version idMapping $
-                Map.map (\oldMapping -> foldl' applyChange oldMapping changes) mappingForUri
+                Map.insert _version id $
+                Map.map ((.) $ foldr (.) id $ fmap toMapping changes) mappingForUri
```

Comment:
oh dang, I think this is in reverse order.

Reply:
Are you saying the old code was incorrect? According to the LSP spec an edit array `[x,y]` is equivalent to first receiving an edit `x` and then an edit `y` so `applyChange (applyChange oldMapping x) y` which is equivalent to `foldl' applyChange oldMapping [x,y]`.

Reply:
I find the `Iso` stuff and the new code quite hard to read. I would prefer if we could stick to the code structure that we’ve had before.